### PR TITLE
feat(drag-and-drop): v0 UI

### PR DESCRIPTION
**Pull request by georgedoescode**
**This PR:** 

- Implements "Drag Previews" for dragged elements. Drag previews render an abstract representation of the dragged element when possible, and default to a simple rectangle for small elements. 
- Adds the ability to opt in to drag and drop using the `data-sanity-draggable` HTML attribute.
- Improves drag insert position calculation, allowing for larger "hit" areas. 
- Adds an "Insert Marker" preview for drag sequences. This is a simple blue bar that appears on element top/bottom for vertical drag groups, or left/right for horizontal drag groups.
- Commits a simple `dnd.md` document to help provide insight into how drag and drop works. This can be refined over time, and help inform our actual documentation.
- Updates the Page Builder demo to allow `FeatureHighlight` to render an array of `CTAs` — this is helpful for demo-ing horizontal drag and drop on simple text nodes.
- Ensures drag sequences start only after a small drag delta on overlay elements.

@rdunk I have not included your `useDragEvents` functionality in this PR, as I thought it may make sense for you to add it after the fact, but let me know if that isn't correct! 

**Next Steps:** 

This PR does not include the mutation logic for reordering, it simply emits events. We need to integrate with Rupert/Cody's work 🚀 

**⚠️ Before Merging ⚠️:** 

- Update commit message to `chore()`

### CODE REVIEW

**Review by stipsan — APPROVED** [review]
Stellar work @georgedoescode!!! 💖  I say we merge as soon as @rdunk gives his approval 👍

File: packages/visual-editing/src/ui/OverlayDragInsertMarker.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  return (
+    <div
+      style={{
+        position: 'absolute',
+        transform: `translate(${x}px, ${y}px)`,
+        width,
+        height,
+        background: 'color-mix(in srgb, #556bfc 95%, transparent)',
```
[the remark this answers is not in the document]
Comment (georgedoescode) [reply packages/visual-editing/src/ui/OverlayDragInsertMarker.tsx:78]: Yesss! 🚀

File: packages/visual-editing/src/ui/OverlayDragPreview.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  height: ${({$height}) => `${$height}px`};
+  z-index: 9999999;
+  border: 1px solid var(--drag-preview-border);
+  opacity: var(--drag-preview-opacity);
+
+  .drag-preview-content-wrapper {
+    position: relative;
+    width: 100%;
+    height: 100%;
+    container-type: inline-size;
```
[the remark this answers is not in the document]
Comment (georgedoescode) [reply packages/visual-editing/src/ui/OverlayDragPreview.tsx:44]: We are using _all_ the features!

File: packages/visual-editing/src/ui/Overlays.tsx [hunk 3]
```diff
@@ -151,6 +159,13 @@ export const Overlays: FunctionComponent<{
     }
   }, [elements, perspective, reportDocuments])
 
+  const updateDragPreviewCustomProps = (rootElement: HTMLElement | null, x: number, y: number) => {
+    if (!rootElement) return
+
+    rootElement.style.setProperty('--drag-preview-x', `${x}px`)
+    rootElement.style.setProperty('--drag-preview-y', `${y - window.scrollY}px`)
```
[the remark this answers is not in the document]
Comment (georgedoescode) [reply packages/visual-editing/src/ui/Overlays.tsx:166]: Great call out!

File: packages/visual-editing/src/util/drag-and-drop.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
+# Sanity Presentation Drag and Drop Pseudo-docs
```
[the remark this answers is not in the document]
Comment (georgedoescode) [reply packages/visual-editing/src/util/drag-and-drop.md:1]: Ah, that's awesome. Will add :)

**Review by stipsan — APPROVED** [review]

**Pull request merged**